api/gin/webServer.go

```diff
@@ -0,0 +1,252 @@
+package gin
+
+import (
+	"context"
+	"net/http"
+	"sync"
+	"time"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/api/address"
+	"github.com/ElrondNetwork/elrond-go/api/block"
+	"github.com/ElrondNetwork/elrond-go/api/gin/disabled"
+	"github.com/ElrondNetwork/elrond-go/api/hardfork"
+	"github.com/ElrondNetwork/elrond-go/api/middleware"
+	"github.com/ElrondNetwork/elrond-go/api/network"
+	"github.com/ElrondNetwork/elrond-go/api/node"
+	"github.com/ElrondNetwork/elrond-go/api/shared"
+	"github.com/ElrondNetwork/elrond-go/api/transaction"
+	valStats "github.com/ElrondNetwork/elrond-go/api/validator"
+	"github.com/ElrondNetwork/elrond-go/api/vmValues"
+	"github.com/ElrondNetwork/elrond-go/api/wrapper"
+	"github.com/ElrondNetwork/elrond-go/config"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/facade"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/gin-contrib/cors"
+	"github.com/gin-contrib/pprof"
+	"github.com/gin-gonic/gin"
+)
+
+var log = logger.GetOrCreate("api/gin")
+
+// ArgsNewWebServer holds the arguments needed to create a new instance of webServer
+type ArgsNewWebServer struct {
+	Facade          shared.ApiFacadeHandler
+	ApiConfig       config.ApiRoutesConfig
+	AntiFloodConfig config.WebServerAntifloodConfig
+}
+
+type webServer struct {
+	sync.RWMutex
```

Comment:
the mutex is not properly protecting the resources.
As I see, the protected resources are facade and httpServer, but there are accesses from exported functions without locking the mutex.

Reply:
You're right. Refactored again. added protection everywhere `httpServer` and `facade` are used